### PR TITLE
Implement the ArrayBuffer.prototype.transfer() tests

### PR DESCRIPTION
**Pull request by afmenez**
I am trying to implement the `ArrayBuffer.prototype.transfer()` tests, but I have some questions:

-Are the tests correct?
-I don't have the results for Safari, can anyone run it?
-I have added significance: 'small' but I am not sure what is the relative importance of this feature;

TIA

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
> -I don't have the results for Safari, can anyone run it?

they are both `false` in Safari 12

**Comment by afmenez** [comment]
Fixed indentation and added Safari results.

File: data-esnext.js [hunk 1]
```diff
@@ -2794,6 +2794,38 @@ exports.tests = [
...
+{
+  name: 'ArrayBuffer.prototype.transfer',
+  category: STAGE2,
+  significance: 'small',
+  spec: 'https://github.com/domenic/proposal-arraybuffer-transfer/',
+  subtests: [
+    {
+      name: 'ArrayBuffer.prototype.transfer()',
+      exec: function () {/*
+        return typeof ArrayBuffer.prototype.transfer === 'function';
```
Comment (ljharb) [inline data-esnext.js]: these just test for existence; we should instead/also be testing the functionality.

In other words, the tests should ensure that it actually truncates and detaches a buffer.

**Review by ljharb — CHANGES_REQUESTED** [review]

File: data-esnext.js (hunk 1 above)
Comment (afmenez) [reply data-esnext.js]: I based my tests on some of the "BigInt" tests, which do the same. The "Metadata reflection API", "zones", "Frozen Realms API" and "Promise.try" tests are also implemented like this.

Comment (ljharb) [reply data-esnext.js]: Then those should have been fleshed out too :-)

**Comment by zloirock** [comment]
Some time ago we had [this](https://github.com/kangax/compat-table/commit/fa5b98ac90a6ad4444140b4b5150a564717c2d9e) test.

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
cc @domenic for correctness of the tests

**Review by chicoxyzzy — APPROVED** [review]

**Comment by afmenez** [comment]
Rebased

**Pull request merged**